play-ebean/src/main/java/play/db/ebean/EBeanComponents.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package play.db.ebean;
+
+
+import play.Configuration;
+import play.Environment;
+import play.api.db.evolutions.DynamicEvolutions;
+import play.db.DBApi;
+import play.inject.ApplicationLifecycle;
+
+/**
+ * Classes for Java compile time dependency injection.
+ */
+public interface EBeanComponents {
+
+    ApplicationLifecycle applicationLifecycle();
+
+    Environment environment();
+
+    Configuration configuration();
```

Comment:
@gmethvin Java sometimes can't handle this like Scala does. For example:

```java
// Consider that ConfigurationComponents was mixed here to provide
// the default implementation
public class MyComponents implements SomeComponents, SomeOtherComponent, ConfigurationComponents {
}

interface SomeComponents {

	String configuration();

	default String environment() {
		return "env" + configuration();
	}
}

interface SomeOtherComponent {
	String configuration();

	default String someOtherComponent() {
		return "someOther" + configuration();
	}
}

interface ConfigurationComponents {
	default String configuration() {
		return "configuration";
	}
}
```
This won't compile because `MyComponents is not abstract and does not override abstract method configuration() in SomeComponents` and, in fact, it does not since the default implementation is not inherited like it looks it should be (I didn't research that deeply to understand what were the reasons but it could be to avoid clash between default methods).

The two possible fixes are:

1. Implement the method directly in `MyComponents`, which is not desirable because we would lose the ability do mix the default implementations because of their requirements. For example, the code below wouldn't be possible:

```java
public class MyComponents extends BuiltInComponentsFromContext implements AhcWSComponents, DBComponents, HttpFiltersComponents {
    ...
}
```

2. Split the components into smaller interfaces and express the requirements using inheritance. As the method comes from the same place (`ConfigurationComponents`), we won't have the conflict. This is half baked now and ideally, for the sake of consistency, we would have an interface for each built-in component. This is something that is becoming more and more clear to me (have smaller interfaces), but I appreciate your input here. ;-)